modules/core/src/main/scala/fs2/kafka/internal/KafkaConsumerActor.scala

```diff
@@ -415,23 +415,21 @@ private[kafka] final class KafkaConsumerActor[F[_], K, V](
     def handleBatch(newRecords: ConsumerRecords): F[Unit] =
       ref.get.flatMap { state =>
         if (state.fetches.isEmpty) {
-          if (newRecords.isEmpty) F.unit
-          else {
-            ref
-              .updateAndGet(_.withRecords(newRecords))
-              .log(StoredRecords(newRecords, _))
-          }
+          ref
+            .updateAndGet(_.withRecords(newRecords))
+            .log(StoredRecords(newRecords, _))
+            .whenA(newRecords.nonEmpty)
         } else {
           val allRecords = state.records combine newRecords
 
-          if (allRecords.nonEmpty) {
+          F.whenA(allRecords.nonEmpty) {
```

Comment:
there is syntax for this, `doSomething.whenA(thing)`. Probably want to check how @vlovgr feels about it, personally I find `F.thing { ... }` a bit hard to read